### PR TITLE
Update to Laravel 5.8

### PR DESCRIPTION
**Pull request by theimerj**
This is my attempt to update Bagisto to Laravel 5.8. I went through the Laravel documentation and tried to satisfy every step in 5.6 to 5.7 upgrade guide as well as in 5.7 to 5.8 upgrade guide. Any help with this is welcome.

### CODE REVIEW

**Comment by prashant-webkul** [comment]
@jyoti-webkul Please test this branch.
@theimerj Thanks for such generous efforts.

**Comment by prashant-webkul** [comment]
@theimerj Please resolve your conflicts. We need our QA to go through your branch.

Regards:
**Prashant.**

File: composer.json [hunk 1]
```diff
@@ -8,30 +8,30 @@
...
+      "dimsav/laravel-translatable": "^9.0",
+      "doctrine/dbal": "2.9.2",
+      "fideloper/proxy": "^4.0",
+      "flynsarmy/db-blade-compiler": "*",
+      "guzzlehttp/guzzle": "~6.0",
+      "intervention/image": "^2.4",
+      "intervention/imagecache": "^2.3",
+      "kalnoy/nestedset": "^4.3",
+      "konekt/concord": "^1.2",
+      "laravel/framework": "5.6.*",
```
Comment (prashant-webkul) [inline composer.json]: Hey?

**Comment by flakerimi** [comment]
> @theimerj
> 
> Getting exception after clicking on following tabs at admin end
> Sales, Catalog, Customers,Promotions, Settings and CMS
> And getting same exception after clicking on Orders at customer end
> 
> **Exception:-**
> 
> Screenshot from 2019-09-12 19-25-14
> 
> **Regards Bhanu**

fire() has become dispatch()

**Comment by theimerj** [comment]
@flakerimi I believe this is resolved by the last commit  https://github.com/bagisto/bagisto/pull/1364/commits/c838cfe92203ae69eee7e82cfc486b94d28b7832 or am I mistaken?

**Comment by bhanu-webkul** [comment]
> @flakerimi I believe this is resolved by the last commit [c838cfe](https://github.com/bagisto/bagisto/commit/c838cfe92203ae69eee7e82cfc486b94d28b7832) or am I mistaken?

@theimerj  Yes, This issue has been fixed.

**Comment by flakerimi** [comment]
Yeah I just saw it, after post.

Best Regards

Flakerim Ismani
cel : +383 49 424 555

**Comment by flakerimi** [comment]
@theimerj I just test it and works fine. While we are here, lets take it to 6.0 LTS. I'm willing to help.
@prashant-webkul, @bhanu-webkul  what do you think?

**Comment by bhanu-webkul** [comment]
> @theimerj
> 
> **Payment Methods** option is not available under **Sales** in **Configure**.
> 
> Screenshot - 2019-09-12T193645 351
> 
> **Regards Bhanu**

@flakerimi  Have you checked this

**Comment by flakerimi** [comment]
I just did and its missing here too. Let me examine why, and will get back with solution, probably.

**Comment by flakerimi** [comment]
@bhanu-webkul @theimerj 
1.When I move Payment/src/Config/system.php data to Shipping/src/Config/system.php it shows up.

2. When is rename 'sales.paymentmethods' key to 'ninja.paymentmethods' its shows as screenshow.



I am out of options why with 'sales' key doesn't work. Any idea?

**Comment by ghermans** [comment]
any update regarding the upgrade?

**Comment by prashant-webkul** [comment]
> @bhanu-webkul @theimerj
> 1.When I move Payment/src/Config/system.php data to Shipping/src/Config/system.php it shows up.
> 
> 1. When is rename 'sales.paymentmethods' key to 'ninja.paymentmethods' its shows as screenshow.
> 
> Screen Shot 2019-09-20 at 16 56 20
> 
> I am out of options why with 'sales' key doesn't work. Any idea?

The configuration file in Payments packages needs to merged with **paymentmethods.php** present in **config** folder. That part of code is present in PaymentServiceProvider and the method needs to mergeConfigFrom(...).

**Comment by rahulshukla-webkul** [comment]
Hi @theimerj 

Kindly resolve conflicts.

**Comment by jitendra-webkul** [comment]
Updated to 6.0

**Pull request closed without merging**